controller/GiUnplannedItem.controller.js

```diff
@@ -0,0 +1,803 @@
+		 * (NOT before the first rendering! onInit() is used for that one!).
+		 * @memberOf smud.org.wmLM01.view.GiUnplannedItem
+		 */
+		//	onBeforeRendering: function() {
+		///	},
+
+		/**
+		 * Called when the View has been rendered (so its HTML is part of the document). Post-rendering manipulations of the HTML could be done here.
+		 * This hook is the same one that SAPUI5 controls get after being rendered.
+		 * @memberOf smud.org.wmLM01.view.GiUnplannedItem
+		 */
+		//	onAfterRendering: function() {
+		//	},
+
+		/**
+		 * Called when the Controller is destroyed. Use this one to free resources and finalize activities.
+		 * @memberOf smud.org.wmLM01.view.GiUnplannedItem
+		 */
+		//onExit: function() {
+		//},
+		liveChangeQty: function(oEvent, quantity) {
+			var qty;
+			if (!oEvent) {
+				qty = quantity;
+			} else {
+				qty = oEvent.getSource().getValue();
+			}
+			var oModel = this.getView().getModel();
+
+			if (oModel.getProperty("/Meins") == "EA") {
+				qty = parseInt(qty).toString();
+				oModel.setProperty("/Menge", qty);
+			}
+
+			var serialNumbers = oModel.getProperty("/serialNumbers");
+			if (qty < 1 || !oModel.getProperty("/serialNumbersRequired")) {
+				snTable.removeAllItems();
+				oModel.setProperty("/serialNumbers", []);
+				serialNumbers = [];
+				return;
+			}
+
+			// Remove extra serial number rows
+			while (qty < serialNumbers.length) {
+				serialNumbers.pop();
+			}
+
+			// Add additional rows
+			while (qty > serialNumbers.length) {
+				serialNumbers.push({
+					Sernr: ""
+				});
+			}
+
+			var colItemTemplate = sap.ui.getCore().byId(this.getView().createId("snColumnListItem"));
+			var tabModel = new sap.ui.model.json.JSONModel(serialNumbers, false);
+			tabModel.setData(serialNumbers, false);
+
+			var context = new sap.ui.model.Context(tabModel, "/");
+			snTable.setBindingContext(context);
+			snTable.setModel(tabModel);
+			snTable.bindItems("/", colItemTemplate);
+		},
+		filterTable: function(oTable, fields, filterString) {
+			// filter list
+			var oBinding = oTable.getBinding("items");
+			if (oBinding) {
+				var oFilters = [];
+				for (var i = 0; i < fields.length; i++) {
+					oFilters.push(new sap.ui.model.Filter(fields[i], sap.ui.model.FilterOperator.Contains, filterString));
+				}
+				var filterObj = new sap.ui.model.Filter(oFilters, false);
+				oBinding.filter(filterObj);
+			} else {
+				oBinding.filter([]);
+			}
+		},
+		liveChangeBin: function(oEvent) {
+			GlobalFuncs.uppercase(oEvent);
+			if (!oEvent.getSource().getValue()) {
+				return;
+			}
+			zController.getView().getModel().setProperty("/Matnr", "");
+			zController.getView().getModel().setProperty("/Verme", "");
+			zController.getView().getModel().setProperty("/Maktg", "");
+			zController.getView().getModel().setProperty("/Meins", "");
+			this.getView().getModel().setProperty("/Lgpla", oEvent.getSource().getValue());

```

Comment:
if you bind the `value` property of the bindings to the view you don't need to set the property in JS